LDEventSource/LDEventSource.m

```diff
@@ -160,53 +154,29 @@ - (void)URLSession:(NSURLSession *)session dataTask:(NSURLSessionDataTask *)data
 - (void)URLSession:(NSURLSession *)session dataTask:(NSURLSessionDataTask *)dataTask didReceiveData:(NSData *)data
 {
     NSString *eventString = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
-    NSArray *lines = [eventString componentsSeparatedByCharactersInSet:[NSCharacterSet newlineCharacterSet]];
-    
-    LDEvent *event = [LDEvent new];
-    event.readyState = kEventStateOpen;
-    
-    for (NSString *line in lines) {
-        if ([line hasPrefix:ESKeyValueDelimiter]) {
-            continue;
-        }
-        
-        if (!line || line.length == 0) {
-            __weak typeof(self) weakSelf = self;
-            dispatch_async(messageQueue, ^{
-                [weakSelf _dispatchEvent:event];
-            });
-            
-            event = [LDEvent new];
-            event.readyState = kEventStateOpen;
-            continue;
-        }
-        
-        @autoreleasepool {
-            NSScanner *scanner = [NSScanner scannerWithString:line];
-            scanner.charactersToBeSkipped = [NSCharacterSet whitespaceCharacterSet];
-            
-            NSString *key, *value;
-            [scanner scanUpToString:ESKeyValueDelimiter intoString:&key];
-            [scanner scanString:ESKeyValueDelimiter intoString:nil];
-            [scanner scanUpToCharactersFromSet:[NSCharacterSet newlineCharacterSet] intoString:&value];
-            
-            if (key && value) {
-                if ([key isEqualToString:LDEventEventKey]) {
-                    event.event = value;
-                } else if ([key isEqualToString:LDEventDataKey]) {
-                    if (event.data != nil) {
-                        event.data = [event.data stringByAppendingFormat:@"\n%@", value];
-                    } else {
-                        event.data = value;
-                    }
-                } else if ([key isEqualToString:LDEventIDKey]) {
-                    event.id = value;
-                    self.lastEventID = event.id;
-                } else if ([key isEqualToString:LDEventRetryKey]) {
-                    self.retryInterval = [value doubleValue];
-                }
-            }
-        }
+    [self.dataAccumulator accumulateDataWithString:eventString];
+    if ([self.dataAccumulator isReadyToParseEvent]) {
+        [self parseEventString:self.dataAccumulator.dataString];
+        [self.dataAccumulator reset];
+    }
+}
+
+- (void)parseEventString:(NSString*)eventString {
+    if (eventString.length == 0) { return; }
+    LDEventParser *parser = [[LDEventParser alloc] init];
+    LDEvent *event = [parser eventFromString:eventString];
+    if (event) {
+        __weak typeof(self) weakSelf = self;
+        dispatch_async(messageQueue, ^{
+            [weakSelf _dispatchEvent:event];
+        });
+        self.lastEventID = event.id;
+    }
+    if (parser.foundRetryInterval > 0.0) {
```

Comment:
It would be good to distinguish between an unset/unchanged retry interval and a retry value of `0.0`. Perhaps use -1 as an unset value?

Reply:
This change is a bit of a divergence from the behavior of the original `eventsource` library and the eventsource spec, but it might be fine if `retryInterval` is never 0. I can think of few real cases where anyone would want a zero retry interval.

Reply:
I used an NSNumber, instead of NSTimeInterval in the parser. The parser leaves the retry interval nil until it finds a `retry` key, and then it stores the value. LDEventSource reads when the retry interval exists, and stores it as before. 

This avoids having to define a constant.